docs/develop/contracts/debug-locally.md

```diff
@@ -0,0 +1,61 @@
+---
+id: debug-locally
+title: Debug Testnet or Mainnet Contract Locally
+sidebar_label: Debug Locally
+---
+
+After contracts are deployed on testnet or mainnet, you might still experience corner case of:
+
+- contract execution logic does not happen in expected way;
+- storage usage is higher than expected;
+- gas cost is higher than expected.
+
+In these cases, you may want to capture the contract state and function call arguments to debug in the local sandbox. You'll make incremental changes, test locally until everything works as expected. You will also likely want to keep the test fixtures to avoid regression in future.
+
+## Obtain States and Other Parameters from Testnet/Mainnet
+
+The following things determined the contract call results:
+
+- Contract code and state;
+- Node's genesis config;
+- Context data when the transaction happens, like current epoch, block height, gas price etc;
+- Caller (signer) of the transaction and arguments.
+
+We'll need to obtain all this information to reproduce the contract call locally. This information can be divide into three categories, each of them is configured in a different way:
+
+- Genesis config
+- State records
+- Parameter provided when sending transaction
+
+### Genesis config
+
+The context data when transaction happens, like current epoch, block height, gas price etc. and testnet/mainnet specific configuration such as block time, gas amount for different kind of actions all come into this category. This parameters can only be set by start a local sandbox node from a modified version of genesis file.
+First thing you need is fetch testnet/mainnet genesis config. This can be obtained by [Genesis Config RPC](https://docs.near.org/docs/api/rpc#genesis-config). Use response file to replace sandbox's genesis.json. Now we have the same configuration as the testnet/mainnet in sandbox. But there's still something different: protocol version, block height and block timestamp. This is because genesis indicates the start of the most recent hard fork of testnet/mainnet, but many blocks and time has passed since the genesis. There's also protocol version upgrades since genesis. If your contract logic doesn't depend on these parameters replay contract method on such parameters, sandbox node should give you accurate logic, gas and storage simulation. However if you do require info like certain epoch and block height, you can modify `"genesis_height"`, `"protocol_version"` on genesis.json.
```

Comment:
```suggestion
The first thing you'll need to do is fetch the `testnet`/`mainnet` genesis config. This can be obtained by the [Genesis Config RPC](https://docs.near.org/docs/api/rpc#genesis-config). Use the response file to replace the sandbox's `genesis.json`. Once complete, you will have the same configuration as `testnet`/`mainnet` in the sandbox aside from the protocol version, block height, and block timestamp. This is because genesis indicates the start of the most recent hard fork of `testnet`/`mainnet` but many blocks have been produced, protocol versions released, and time has passed the blockchain genesis. If your contract logic doesn't depend on these parameters the sandbox node should give you accurate logic, gas, and storage simulation. However, if you do require info like certain epoch and block height details you can modify `"genesis_height"` and `"protocol_version"` on your `genesis.json` file.
```